rust/operator-binary/src/hdfs_controller.rs

```diff
@@ -349,22 +349,43 @@ pub async fn reconcile_hdfs(hdfs: Arc<HdfsCluster>, ctx: Arc<Ctx>) -> HdfsOperat
 
             let rolegroup_ref = hdfs.rolegroup_ref(role_name, rolegroup_name);
 
-            let rg_service =
-                rolegroup_service(&hdfs, &role, &rolegroup_ref, &resolved_product_image)?;
+            // The "met_builder" variable is needed to keep the rust compiler happy.
+            // Without it, an error is thrown that the "metadata" variable is dropped
+            // while still being in use.
+            let mut met_builder = ObjectMetaBuilder::new();
+            let metadata = met_builder
```

Comment:
Thanks for the comment! First off there is a typo, but I personally have the feeling this is pretty normal stuff and the documentation might mislead readers that this situation is different (I have to admit I didn't understood it).

We can also shadow `metadata` here

```suggestion
            let mut metadata = ObjectMetaBuilder::new();
            let metadata = metadata
```

Reply:
I don't understand the behavior either but if remove line 352 and change line 353 with:
```
            let mut metadata = ObjectMetaBuilder::new()
```
you'll get
```
error[E0716]: temporary value dropped while borrowed
   --> rust/operator-binary/src/hdfs_controller.rs:356:32
    |
356 |             let mut metadata = ObjectMetaBuilder::new()
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use
...
370 |                 .context(ObjectMetaSnafu)?;
    |                                           - temporary value is freed at the end of this statement
371 |
372 |             let rg_service = rolegroup_service(&hdfs, metadata, &role, &rolegroup_ref)?;
    |                                                       -------- borrow later used here
    |
    = note: consider using a `let` binding to create a longer lived value
```

Reply:
I didn't apply your suggestion, instead I tried to clarify the comment too: https://github.com/stackabletech/hdfs-operator/pull/534/commits/bb2a85f137f66c3cef2ed2c1e3092854ba41fa84

Reply:
Perfectly fine, thanks!